src/Component/src/Symfony/Request/State/Provider.php

```diff
@@ -59,14 +61,28 @@ public function provide(Operation $operation, Context $context): object|array|nu
                 $defaultMethod = 'findById';
             }
 
-            $method = $operation->getRepositoryMethod() ?? $defaultMethod;
+            $customMethod = $operation->getRepositoryMethod();
+            $method = $customMethod ?? $defaultMethod;
 
             if (!$this->locator->has($repository)) {
-                throw new \RuntimeException(sprintf('Repository "%s" not found on operation "%s"', $repository, $operation->getName() ?? ''));
+                throw new RuntimeException(sprintf('Repository "%s" not found on operation "%s".', $repository, $operation->getName() ?? ''));
             }
 
+            /** @var object $repositoryInstance */
             $repositoryInstance = $this->locator->get($repository);
 
+            if (
+                !str_starts_with($method, 'find') &&  // magic calls on Doctrine repositories
```

Comment:
> // magic calls on Doctrine repositories

That's not always true, I can have methods like `findOneByAcme`, which doesn't have any magic.

Reply:
That's right, and not always "find" are magic. 
We can replace the text with "To allow magic calls on Doctrine repositories methods".